build.sbt

```diff
@@ -113,32 +125,46 @@ lazy val `scalafix-sbt` = project
   )
   .enablePlugins(BuildInfoPlugin)
 
-lazy val testkit = project
-  .configure(setId)
-  .settings(
-    isFullCrossVersion,
+val testkit = MultiScalaProject(
+  "scalafix-testkit",
+  _.settings(
+    // isFullCrossVersion,
     libraryDependencies ++= Seq(
       semanticdb,
       ammonite,
       googleDiff,
       scalatest.value
     )
-  )
-  .dependsOn(
-    coreJVM,
-    reflect
-  )
+  ))
 
-lazy val testsShared = project
-  .in(file("scalafix-tests/shared"))
-  .settings(
+lazy val testkit211 = testkit(scala211, _.dependsOn(core211JVM, reflect211))
+lazy val testkit212 = testkit(scala212, _.dependsOn(core212JVM, reflect212))
+
+def TestProject(sub: String, configure: Project => Project): MultiScalaProject =
+  MultiScalaProject(
+    s"scalafix-tests-$sub",
+    file(s"scalafix-tests/$sub"),
+    configure)
+
+val testsShared = TestProject(
+  "shared",
+  _.settings(
     semanticdbSettings,
     noPublish
-  )
+  ))
 
-lazy val testsInput = project
-  .in(file("scalafix-tests/input"))
-  .settings(
+lazy val testsShared211 = testsShared(scala211)
+lazy val testsShared212 = testsShared(scala212)
+
+lazy val inputOutputSetting = Seq(
```

Comment:
Custom settings should be in ScalafixBuild.scala, build.sbt should only contain project definitions.